kernel/src/common/copy_slice.rs

```diff
@@ -0,0 +1,50 @@
+//! A wrapper around an immutable slice of `Copy` elements
+//! that provides access via a mutable slice
+
+use core;
+
+pub struct CopySlice<'a, T: 'a + Copy>{
+    ptr: *mut T,
+    len: usize,
+    _phantom: core::marker::PhantomData<&'a T>,
+}
+
+impl<'a, T: Copy> CopySlice<'a, T> {
+    pub fn new(buf: &'a [T]) -> CopySlice<'a, T> {
+        CopySlice{
+            ptr: buf.as_ptr() as *mut T,
+            len: buf.len(),
+            _phantom: core::marker::PhantomData,
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    pub fn as_mut(&self) -> &'a mut [T] {
+        unsafe {
+            core::slice::from_raw_parts_mut(self.ptr, self.len)
+        }
+    }
+
+    pub fn as_slice(&self) -> &'a [T] {
+        unsafe {
+            core::slice::from_raw_parts(self.ptr, self.len)
+        }
+    }
+}
+
+pub fn static_mut_bytes_8() -> &'static mut [u8] {
+    unsafe {
```

Comment:
I think this will only do anything reasonable if it's only ever called once, in which case the function should really be marked unsafe. Basically, the statically allocated slice is always the same one.